mono/metadata/threads.c

```diff
@@ -338,6 +341,8 @@ mono_thread_set_interruption_requested (MonoInternalThread *thread)
 			new_state = old_state | INTERRUPT_ASYNC_REQUESTED_BIT;
 	} while (InterlockedCompareExchangePointer ((volatile gpointer)&thread->thread_state, (gpointer)new_state, (gpointer)old_state) != (gpointer)old_state);
 
+	InterlockedIncrement (&thread_interruption_requested);
```

Comment:
This depends on the convention that we use for thread_interruption_requested. We can either not have thread_int_req bumped for async exceptions that are deferred inside an abort protected block (in which case we would to that check and we would bump thread_int_req when exiting the block) or just bump it as soon as we've set the interruption bit (as is in this version of the code) (in this case the backend will see that there is an interruption requested and check the thread state to execute it, up until it exits the abort protected region, and can finally clear the interruption and decrement thead_int_req)